src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/SqlBatch.cs

```diff
@@ -0,0 +1,234 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Data;
+using System.Data.Common;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Data.Common;
+
+namespace Microsoft.Data.SqlClient
+{
+#pragma warning disable CS1591 // Missing XML comment for publicly visible type or member
+    public class SqlBatch : DbBatch
+    {
+        private SqlCommand _batchCommand;
+        private List<SqlBatchCommand> _commands;
+        private DbBatchCommandCollection _providerCommands;
+
+        public SqlBatch()
+        {
+            _batchCommand = new SqlCommand();
+        }
+
+        public SqlBatch(SqlConnection connection, SqlTransaction transaction = null)
```

Comment:
FWIW, both Npgsql and MySqlConnector make the first argument (`connection`) nullable as well.

https://github.com/npgsql/npgsql/blob/8d6da14034aedb88ec5ad98474a99fa38e670535/src/Npgsql/NpgsqlBatch.cs#L99
https://github.com/mysql-net/MySqlConnector/blob/12d6f818a94f4006ddaaa00ecf5592205408975c/src/MySqlConnector/MySqlBatch.cs#L74

Reply:
Done.